services-custom/dynamodb-enhanced/src/test/java/software/amazon/awssdk/enhanced/dynamodb/functionaltests/document/APISurfaceAreaReview.java

```diff
@@ -0,0 +1,401 @@
+                                                     .addNumber("numberKey", 2.0)
+                                                     .addSdkBytes("sdkByte", SdkBytes.fromUtf8String("a"))
+                                                     .addBoolean("booleanKey", true)
+                                                     .addJson("jsonKey", "{\"1\": [\"a\", \"b\", \"c\"],\"2\": 1}")
+                                                     .addStringSet("stingSet",
+                                                                   Stream.of("a", "b", "c").collect(Collectors.toSet()))
+
+                                                     .addNumberSet("numberSet", Stream.of(1, 2, 3, 4).collect(Collectors.toSet()))
+                                                     .addSdkBytesSet("sdkByteSet",
+                                                                     Stream.of(SdkBytes.fromUtf8String("a")).collect(Collectors.toSet()))
+                                                     .build();
+
+        /**
+         *{
+         *    "HashKey":"abcdefg123",
+         *    "nullKey":null,
+         *    "numberKey":2.0,
+         *    "sdkByte":"a",
+         *    "booleanKey":true,
+         *    "jsonKey":{
+         *       "1":[
+         *          "a",
+         *          "b",
+         *          "c"
+         *       ],
+         *       "2":1
+         *    },
+         *    "stingSet":[
+         *       "a",
+         *       "b",
+         *       "c"
+         *    ],
+         *    "numberSet":[
+         *       1,
+         *       2,
+         *       3,
+         *       4
+         *    ],
+         *    "sdkByteSet":[
+         *       "a"
+         *    ]
+         * }
+
+         */
+
+        assertThat(simpleDoc.toJson()).isEqualTo("{\"HashKey\": \"abcdefg123\",\"nullKey\": null,\"numberKey\": 2.0,"
+                                                            + "\"sdkByte\": \"a\",\"booleanKey\": true,\"jsonKey\": {\"1\": "
+                                                            + "[\"a\", \"b\", \"c\"],\"2\": 1},\"stingSet\": [\"a\", \"b\", "
+                                                            + "\"c\"],\"numberSet\": [1, 2, 3, 4],\"sdkByteSet\": [\"a\"]}");
+
+
+        assertThat(simpleDoc.isPresent("HashKey")).isTrue();
+        // No Null pointer or doesnot exist is thrown
+        assertThat(simpleDoc.isPresent("HashKey2")).isFalse();
+        assertThat(simpleDoc.getString("HashKey")).isEqualTo("abcdefg123");
+        assertThat(simpleDoc.isNull("nullKey")).isTrue();
+
+        assertThat(simpleDoc.getSdkNumber("numberKey")).isNotEqualTo(2.0);
+        assertThat(simpleDoc.getSdkNumber("numberKey").doubleValue()).isEqualTo(2.0);
+
+        assertThat(simpleDoc.getSdkBytes("sdkByte")).isEqualTo(SdkBytes.fromUtf8String("a"));
+        assertThat(simpleDoc.getBoolean("booleanKey")).isTrue();
+        assertThat(simpleDoc.getJson("jsonKey")).isEqualTo("{\"1\": [\"a\", \"b\", \"c\"],\"2\": 1}");
```

Comment:
Sorry it's not clear from the example, how is this different/better for JSON than `add/getString()`?